pkg/gpu/ebpf/c/runtime/gpu.c

```diff
@@ -90,6 +90,51 @@ int BPF_UPROBE(uprobe__cudaLaunchKernel, const void *func, __u64 grid_xy, __u64
     return 0;
 }
 
+
+SEC("uprobe/cuLaunchKernel")
+int BPF_UPROBE(uprobe__cuLaunchKernel, const void *func, __u32 grid_x, __u32 grid_y, __u32 grid_z, __u32 block_x, __u32 block_y) {
+    cuda_kernel_launch_t launch_data = { 0 };
+    long read_ret = 0;
+    __u64 shared_mem = 0;
+    __u64 stream = 0;
+    __u32 block_z = 0;
+
+    block_z = PT_REGS_USER_PARM7(ctx, read_ret);
+    if (read_ret < 0) {
```

Comment:
If you create a WITH_TELEMETRY variant for this macro, you can remove the `read_ret` check and get telemetry without bpf debugging enabled.